ludwig/utils/misc_utils.py

```diff
@@ -216,3 +216,20 @@ def get_file_names(output_directory):
     model_dir = os.path.join(output_directory, 'model')
 
     return description_fn, training_stats_fn, model_dir
+
+
+def check_which_model_definition(model_definition, model_definition_file):
+    # check for model_definition and model_definition_file
+    if model_definition is None and model_definition_file is None:
+        raise ValueError(
+            'Either model_definition of model_definition_file have to be'
+            'not None to initialize a LudwigModel'
+        )
+    if model_definition is not None and model_definition_file is not None:
+        raise ValueError(
+            'Only one between model_definition and '
+            'model_definition_file can be provided'
+        )
+    if not model_definition:
```

Comment:
If the user provided `model_definition` as a string on the command line, we need to load it as a dict, right?  Otherwise, how do we disambiguate the model filename from the model definition after this function is called?

Reply:
You are right but there's no need for it because the model_definition parameter in args has type `yaml.save_load` which parses the string as yaml automatically and the values of args.model_definition is the parsed dict

Reply:
Ah, I see.  Makes sense!